src/psyclone/tests/parse/algorithm_test.py

```diff
@@ -0,0 +1,295 @@
+# pylint: disable=too-few-public-methods
+
+# class parser() tests
+
+
+def test_parser_parse(tmpdir):
+    '''Test that if no relevant code is found in the algorithm file then
+    the appropriate exception is raised.
+
+    '''
+    tmp = Parser()
+    filename = str(tmpdir.join("empty.f90"))
+    ffile = open(filename, "w")
+    ffile.write("")
+    ffile.close()
+    with pytest.raises(ParseError) as excinfo:
+        _ = tmp.parse(filename)
+    assert ("Program, module, function or subroutine not found in parse tree "
+            "for file") in str(excinfo.value)
+
+
+def test_parser_createinvokecall():
+    '''Test that if an argument to an invoke call is not what is expected
+    then the appropriate exception is raised.
+
+    '''
+    from fparser.two.Fortran2003 import Call_Stmt
+    from fparser.two.parser import ParserFactory
+    _ = ParserFactory().create(std="f2003")
+    statement = Call_Stmt("call invoke(0.0)")
+    tmp = Parser()
+    with pytest.raises(ParseError) as excinfo:
+        _ = tmp.create_invoke_call(statement)
+    assert (
+        "Expecting argument to be of the form 'name=xxx' or a Kernel call "
+        "but found '0.0' in file 'None'.") in str(excinfo.value)
+
+
+def test_parser_updateargtomodulemap_invalid():
+    '''Test that if the statement argument to the
+    update_arg_to_module_map method is not a use statement that the
+    appropriate exception is raised.'''
+    tmp = Parser()
+    with pytest.raises(InternalError) as excinfo:
+        tmp.update_arg_to_module_map("invalid")
+    assert "Expected a use statement but found instance of" \
+        in str(excinfo.value)
+
+# function parse_fp2() tests
+
+
+def test_parsefp2_invalid_file(tmpdir):
+    '''Test that if there is an error finding the input file specified in
+    filename argument then an exception is raised in the expected way.
+
+    '''
+    with pytest.raises(ParseError) as excinfo:
+        _ = parse_fp2(str(tmpdir.join("does_not_exist.f90")))
+    assert "Failed to parse file" in str(excinfo.value)
+    assert "Error returned was ' [Errno 2] No such file or directory" \
+        in str(excinfo.value)
+
+
+def test_parsefp2_invalid_fortran(tmpdir):
+    '''Test that if the Fortran contained in the file specified in the
+    filename argument then an exception is raised in the expected
+    way. *** Create the parse_tree in-place rather than running
+    PSyclone. Once created make the parse_tree content invalid using
+    monkeypatch.
+
+    '''
+    my_file = str(tmpdir.join("invalid.f90"))
+    ffile = open(my_file, "w")
+    ffile.write("invalid Fortran code")
+    ffile.close()
+    with pytest.raises(ParseError) as excinfo:
+        _ = parse_fp2(my_file)
+    assert "Syntax error in file" in str(excinfo.value)
+
+
+# function get_invoke_label() tests
+
+
+def test_getinvokelabel_invalid_tree():
+    '''Test that if the parse tree argument is not an Actual_Arg_Spec
+    then an exception is raised in the expected
```

Comment:
Strange formatting. Please bring "way." up to the line above.

Reply:
Done. And one other further down the file too.